### PR TITLE
chore(react-composition): migrate elements utilities to ui-react-core

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Migrate `elements` utilities to _@aws-amplify/ui-react-core_:
- expose `elements` utilities from _@aws-amplify/ui-react-core_
- update `withBaseElementProps` callback `defaultProps` behavior
- remove `useElement` in favor of direct `elements` lookup via `React.useContext`
- add additional keys to `ElementDisplayName`

#### Issue #, if available
NA

#### Description of how you validated changes
Manual tests, unit tests
#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] `yarn test` passes and tests are updated/added
- [x] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-core/src/elements/types.ts [hunk 1]
```diff
@@ -15,11 +17,40 @@ export type BaseElement<T = {}, K = {}> = React.ForwardRefExoticComponent<
   React.PropsWithoutRef<T> & React.RefAttributes<K>
 >;
 
+type ListItemElementSubType = 'Menu' | 'Nav' | 'List';
+type ListItemElementDisplayName = `${ListItemElementSubType}Item`;
```
Comment (hbuchel) [inline packages/react-core/src/elements/types.ts:21]: Does ListItemElement need sub types? Essentially all of these end up being an `<li>` I think. A menu item (role="menuitem") would end up going on a Button that is a child of role="menu".

**Review by hbuchel — COMMENTED** [review]
Just a question about the ListItemElementSubType

**Review by thaddmt — APPROVED** [review]

File: packages/react-core/src/elements/types.ts (hunk 1 above)
Comment (calebpollman) [reply packages/react-core/src/elements/types.ts:21]: Was thinking more along the lines of differentiation between use cases but am good with removing the sub types, thoughts?

Comment (hbuchel) [reply packages/react-core/src/elements/types.ts:21]: Yeah I think I'm good with removing the ListItemElement subtypes; the ones for ListElement and TableElement make sense.

**Review by hbuchel — APPROVED** [review]

File: packages/react-core/src/elements/ElementsContext.tsx [hunk 2]
```diff
@@ -71,10 +71,3 @@ export function ElementsProvider<T extends Elements>({
 }): React.JSX.Element {
   return <ElementsContext.Provider {...props} value={elements} />;
 }
-
-export const useElement = <T extends keyof Elements>(
-  name: T
-): Elements[T] | undefined => {
-  const context = React.useContext(ElementsContext);
-  return context?.[name];
-};
```
Comment (thaddmt) [inline packages/react-core/src/elements/ElementsContext.tsx:73]: Makes sense to remove so we don't try to use it unnecessarily

**Review by thaddmt — APPROVED** [review]

File: packages/react-core/src/elements/ElementsContext.tsx (hunk 2 above)
Comment (calebpollman) [reply packages/react-core/src/elements/ElementsContext.tsx:73]: That was the intent 🤠

**Pull request merged**